app/javascript/listings/listings.jsx

```diff
@@ -282,29 +306,38 @@ export class Listings extends Component {
     window.history.replaceState(null, null, newLocation);
   };
 
+  /**
+   * Call search API for ClassifiedListings
+   *
+   * @param {string} query - The search term
+   * @param {string} tags - The tags selected by the user
+   * @param {string} category - The category selected by the user
+   * @param {string} slug - The listing's slug
+   *
+   * @returns {Promise} A promise object with response formatted as JSON.
+   */
   listingSearch(query, tags, category, slug) {
     const t = this;
-    const { index, page, listings } = t.state;
-    const filterObject = { tagFilters: tags, hitsPerPage: 75, page };
-    if (category.length > 0) {
-      filterObject.filters = `category:${category}`;
-    }
-    index.search(query, filterObject).then(function searchDone(content) {
-      const fullListings = listings;
-      content.hits.forEach(listing => {
-        if (listing.bumped_at) {
-          if (!listings.map(l => l.id).includes(listing.id)) {
-            fullListings.push(listing);
-          }
-        }
-      });
+    const { page, listings } = t.state;
+    const dataHash = {
+      category,
+      classified_listing_search: query,
+      page,
+      per_page: LISTING_PAGE_SIZE,
+      tags,
+    };
```

Comment:
This is much cleaner now. The params cleanup logic has been moved to the `search_controller`.